src/helics/common/SmallBuffer.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright (c) 2017-2020,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable
+Energy, LLC.  See the top-level NOTICE for additional details. All rights reserved.
+SPDX-License-Identifier: BSD-3-Clause
+*/
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <string>
+#include <string_view>
+#include <utility>
+
+namespace helics {
+class SmallBuffer {
+  public:
+    SmallBuffer() noexcept: heap(buffer.data()) {}
+
+    SmallBuffer(const SmallBuffer& sb);
+    SmallBuffer(SmallBuffer&& sb) noexcept;
+    SmallBuffer(std::string_view val);
+    SmallBuffer(const void* data, size_t size);
+    ~SmallBuffer();
+    SmallBuffer& operator=(const SmallBuffer& sb);
+    SmallBuffer& operator=(SmallBuffer&& sb) noexcept;
+    SmallBuffer& operator=(std::string_view val);
+    /** return a pointer to the data location*/
+    std::byte* data() const { return heap; }
+    /** get the start of the data*/
+    std::byte* begin() { return heap; }
```

Comment:
`begin()` and `end()` may be more function if return type is `std::iterator<std::input_iterator_tag, std::Byte>` rather than a direct pointer, as this will enable range-based iteration across SmallBuffer.

Reply:
I am pretty sure pointers are considered iterators for range based for loops.  and some of the tests use a range based for loop in the test.    So my inclination is to leave this as a pointer since then begin and end can be used in the assign of another Small buffer, and I don't want to make the assign a template right now.